src/librustc/middle/resolve_lifetime.rs

```diff
@@ -917,6 +917,16 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                 }
             }
             self.insert_lifetime(lifetime_ref, def);
+        } else if lifetime_ref.is_static() {
+            // The only way we end up here is if there is a static lifetime bound (due to early
+            // return on `visit_lifetime()`)
+            self.insert_lifetime(lifetime_ref, Region::Static);
+            let full_span = mk_sp(context_ref.span.lo, lifetime_ref.span.hi);
+            self.sess.struct_span_warn(full_span,
+                &format!("unnecessary lifetime parameter `{}`", context_ref.name))
+                .help(&format!("you can use a `'static` lifetime directly, in place of \
+                                the lifetime parameter `{}`", context_ref.name))
```

Comment:
I would reword: ``you can use the lifetime `'static` directly in place of `{}` ``.